pkg/kube/krt/collection.go

```diff
@@ -111,16 +107,25 @@ func (i dependencyState[I]) changedInputKeys(sourceCollection collectionUID, eve
 		// inefficient, especially when the dependency changes frequently and the collection is large.
 		// Where possible, we utilize the reverse-indexing to get the precise list of potentially changed objects.
 		foundAny := false
-		for _, idxTypes := range allIndexedDependencyTypes {
-			ekey := extractorKey{uid: sourceCollection, typ: idxTypes}
+
+		// find all the indexes/collections we depend on
```

Comment:
I spot checked this and the largest `N` I saw in a cluster that had waypoints, authz, ambient enabled was 6

Reply:
Does multicluster #55552 specifically break this assumption?

Reply:
Most likely it multiplies it by the number of clusters? I think that's still pretty reasonable scale. .

Reply:
a slight optimization here would be making `indexedDependenciesExtractor` into a map[collectionUID]map[extractorKey]extractor`.